pkg/collector/corechecks/cluster/topologycollectors/node_collector.go

```diff
@@ -99,8 +94,10 @@ func (nc *NodeCollector) nodeToStackStateComponent(node v1.Node) *topology.Compo
 	var instanceID string
 	if len(node.Spec.ProviderID) > 0 {
 		instanceID = extractInstanceIDFromProviderID(node.Spec)
-		identifiers = append(identifiers, fmt.Sprintf("urn:host:/%s", instanceID))
+	} else {
+		instanceID = node.Name
```

Comment:
If `node.Name` is always defined, maybe it worth to always add identifier based on it? That will make result set of identifiers more straightforward. So instead of guessing why you get different identifier, a person will only be guessing one one of identifers (based of ProviderID) is missing.

Reply:
The problem with using the name is that you can have other two nodes with the same node from different environments, if so they will merge. That's what I understood from @zandrewitte. I don't know what is the chance of that, to be honest. Maybe we can discuss it with Mark?